SmartEVSE-3/src/evse.cpp

```diff
@@ -384,7 +390,7 @@ void BlinkLed(void * parameter) {
                 BluePwm = 0;
             }
 
-        } else if (Access_bit == 0) {                                            // No Access, LEDs off
+        } else if (Access_bit == 0 || State == STATE_MODEM_DENIED) {                                            // No Access, LEDs off
```

Comment:
Every edge case in the SmartEVSE code needs to be hacked in somewhere in a decision tree.  It lacks a bit of architecture and design I think.
You see this example happening a lot, not just in your code, also in mine and others. Every edge case needs another if/else change.

Reply:
Exactly. There's also a lot of repeated code related to maintaining states. At least in case of your recent additions, it was very well documented. For other features, it was less clear what it does and why. I suppose we're at a point where I'd rather not touch what's already there, but I still think it'd be a great idea to organise it a bit better.